### PR TITLE
feat: 🍰 Redesign Registration Process Frontend

### PR DESCRIPTION
**Pull request by Tirokk**
## 🍰 Pullrequest

Redesign the frontend of the registration process.

### Issues

- relates #4092

### Todo

- [ ] a lot

### CODE REVIEW

**Comment by Tirokk** [comment]
I added more functionality. @Mogge

File: webapp/components/Registration/RegistrationItemEnterNonce.vue [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        },
+      })
+    },
+    async handleInput() {
+      this.sendValidation()
+    },
+    async handleInputValid() {
+      this.sendValidation()
+    },
+    handleSubmitVerify() {},
```
Comment (Tirokk) [inline webapp/components/Registration/RegistrationItemEnterNonce.vue:93]: A database request would be nice here as for the invite.
Or what you think. @Mogge @ulfgebhardt

**Review by Tirokk — COMMENTED** [review]
Please do a pre-review. @Mogge @ulfgebhardt 

This is my own pre-review to give you some hints about questions and leftovers and have some reminders for me.

- I haven't done translations in case we will have general changes in the texts …
- Comments marked with ‚Wolle‘ have do be proven and probably deleted.
But this is not sure at the moment.

File: webapp/components/ComponentSlider/ComponentSlider.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    async onNextClick() {
+      let success = true
+      if (this.sliderData.sliders[this.sliderIndex].button.sliderCallback) {
+        success = await this.sliderData.sliders[this.sliderIndex].button.sliderCallback()
+      }
+      success = success && this.sliderData.sliders[this.sliderIndex].button.callback(success)
+      if (success && this.sliderIndex < this.sliderData.sliders.length - 1) {
+        this.sliderData.sliderSelectorCallback(this.sliderIndex + 1)
+      }
+    },
```
Comment (ulfgebhardt) [inline webapp/components/ComponentSlider/ComponentSlider.vue:71]: This whole function seems a bit omplicated. Why three functions `sliderCallback`, `callback` and `sliderSelectorCallback` ?

File: webapp/components/Registration/RegistrationItemCreateUserAccount.vue [hunk 3]
```diff
@@ -0,0 +1,321 @@
+<template>
```
Comment (ulfgebhardt) [inline webapp/components/Registration/RegistrationItemCreateUserAccount.vue:1]: I really would love to see this page split into several pages as suggested by the design. What do you think?

-> Furthmore I believe that those componenents should be named something like `RegisterationSlide` or `RegistrationPage` not `RegistrationItem`

File: webapp/components/Registration/RegistrationSlider.vue [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+      </template>
+
+      <component-slider :sliderData="sliderData">
+        <template #header>
+          <ds-heading size="h2">
+            {{ $t('components.registration.signup.title', metadata) }}
+          </ds-heading>
+        </template>
+
+        <template v-if="['invite-code'].includes(registrationType)" #enter-invite>
```
Comment (ulfgebhardt) [inline webapp/components/Registration/RegistrationSlider.vue:17]: shouldn't it be `registrationType.includes('invite-code')` ?

File: webapp/components/Registration/RegistrationSlider.vue [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+          nonce: null,
+          name: null,
+          password: null,
+          passwordConfirmation: null,
+          about: null,
+          termsAndConditionsConfirmed: null,
+          dataPrivacy: null,
+          minimumAge: null,
+          noCommercial: null,
+          noPolitical: null,
```
Comment (ulfgebhardt) [inline webapp/components/Registration/RegistrationSlider.vue:167]: maybe we can put that into each slider and just have a array of sliderdata objects. This is not needed for this PR tho - just as reminder for refactoring

**Review by ulfgebhardt — COMMENTED** [review]
Great work - code looks pretty clean. Please see my comments - but I believe we can push that through next week <3

File: webapp/components/ComponentSlider/ComponentSlider.vue (hunk 2 above)
Comment (Tirokk) [reply webapp/components/ComponentSlider/ComponentSlider.vue:71]: We invent a more general slider component and therefore I guess it is not always clear if decisions about what will happen on a button click can always only be done in the deepest level.
Therefore I implement a mechanism to have the possibility to decide at top-level as well or in combination. @ulfgebhardt

File: webapp/components/Registration/RegistrationItemCreateUserAccount.vue (hunk 3 above)
Comment (Tirokk) [reply webapp/components/Registration/RegistrationItemCreateUserAccount.vue:1]: I have not to much heart in that it must be this big. And the base card could have a bit more width to give the user a better overview.
This way it is split into kind of logical steps …

I would like to rename the components as well. 😍👍🏼

File: webapp/components/Registration/RegistrationSlider.vue (hunk 4 above)
Comment (Tirokk) [reply webapp/components/Registration/RegistrationSlider.vue:17]: I’ve used the same style as for the other slider components to have a better recognition that’s about the same thing. Could be for sure as well:
```suggestion
        <template v-if="registrationType === 'invite-code'" #enter-invite>
```
 As far I've understand the array comes before the dot selector and `registrationType` is not an array …

File: webapp/components/Registration/RegistrationSlider.vue (hunk 5 above)
Comment (Tirokk) [reply webapp/components/Registration/RegistrationSlider.vue:167]: I think it's not needed for code functionality anyways. For me it has the functionality to give an overview about all the data to collect at the top-level.

**Review by Mogge — APPROVED** [review]

**Pull request merged**